deploy/aws/lambda-ext/main.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021-2025 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/cerbos/cerbos/internal/config"
+	"github.com/cerbos/cerbos/internal/observability/logging"
+	runutils "github.com/cerbos/cerbos/internal/run"
+	"github.com/cerbos/cerbos/internal/server"
+	"github.com/cerbos/cerbos/pkg/cerbos"
+	"github.com/sourcegraph/conc/pool"
+	"go.uber.org/zap"
+)
+
+func main() {
+	ctx, stopFunc := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+	defer stopFunc()
+
+	configPath := os.Getenv("CERBOS_CONFIG")
+	if configPath == "" {
+		configPath = "/conf.yml"
+	}
+
+	logLevel := os.Getenv("CERBOS_LOG_LEVEL")
+	if logLevel == "" {
+		logLevel = "info"
+	}
+
+	logging.InitLogging(ctx, logLevel)
+	defer zap.L().Sync() //nolint:errcheck
+
+	runtimeAPI := os.Getenv("AWS_LAMBDA_RUNTIME_API")
+
+	log := zap.L().Named("lambda-ext")
+	if runtimeAPI == "" {
+		log.Error("AWS_LAMBDA_RUNTIME_API env var not set, exiting")
+		exit2()
+	}
+
+	p := pool.New().WithContext(ctx).WithCancelOnError().WithFirstError()
+	p.Go(func(ctx context.Context) error {
+		return cerbos.Serve(ctx,
+			cerbos.WithConfigFile(configPath),
+			cerbos.WithLogLevel(cerbos.LogLevel(logLevel)),
+		)
+	})
+	if err := waitForReady(ctx); err != nil {
+		log.Error("Readiness check failed", zap.Error(err))
+		exit2()
+	}
+	p.Go(func(ctx context.Context) error {
```

Comment:
This should go inside a `p.Go` as well because if the cerbos process started above fails for some reason, it won't be aware of it and will hang forever until the entire process is killed by the runtime or by the user.

```suggestion
	p.Go(func(ctx context.Context) error {
	    if err := waitForReady(ctx); err != nil {
		    log.Error("Readiness check failed", zap.Error(err))
		    return fmt.Errorf("readiness check failed: %w", err)
	    }
```